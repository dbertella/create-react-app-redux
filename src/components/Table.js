import React, { PropTypes } from 'react';
import Tr from './Tr';

const renderHeading = (artist) => <Tr info={Object.keys(artist)} />;

const renderList = (artistList) => artistList.map((artist, i) =>
    <Tr key={i} info={Object.keys(artist).map(key => artist[key])} />);

const Table = (props) => {
  const { artistList } = props;
  return (
    <table>
      <thead>
        {
          artistList.length > 0 && renderHeading(artistList[0])
        }
      </thead>
      <tbody>
        {
          artistList.length > 0 && renderList(artistList)
        }
      </tbody>
    </table>
  );
};

Table.propTypes = {
  artistList: PropTypes.array.isRequired,
}

export default Table;
