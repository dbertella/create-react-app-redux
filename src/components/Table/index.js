import React, { PropTypes } from 'react';
import Tr from '../Tr';

const dataToRender = (artist) => {
  return Object.keys(artist).sort(); //.filter(key => key !== 'longitude').filter(key => key !== 'latitude');
}

const renderHeading = (artist) => <Tr info={dataToRender(artist)} />;

const renderList = (artistList) => artistList.map((artist, i) =>
    <Tr key={i} info={dataToRender(artist).map(key => artist[key])} />);

const Table = (props) => {
  const { artistList } = props;
  return (
    <table>
      <thead>
        {
          artistList.length > 0 && renderHeading(artistList[0])
        }
      </thead>
      <tbody>
        {
          artistList.length > 0 && renderList(artistList)
        }
      </tbody>
    </table>
  );
};

Table.propTypes = {
  artistList: PropTypes.array.isRequired,
}

export default Table;
