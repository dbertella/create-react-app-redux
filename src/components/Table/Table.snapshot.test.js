import React from 'react';
import Table from './';
import renderer from 'react-test-renderer';

const artistList = [{
  "uuid": "f853578c-fc0f-4e65-81b8-566c5dffa35a",
  "gender": "F",
  "age": 39,
  "longitude": "-0.09998975",
  "rate": 14.21,
  "latitude": "51.75436293"
}, {
  "uuid": "fe2c3195-aeff-487a-a08f-e0bdc0ec6e9a",
  "gender": "M",
  "age": 66,
  "longitude": "0.18228006",
  "rate": 39.5,
  "latitude": "51.74640997"
}];

it('initial render', () => {
  const component = renderer.create(
    <Table artistList={artistList} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
