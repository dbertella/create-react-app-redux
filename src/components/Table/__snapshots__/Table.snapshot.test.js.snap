exports[`test initial render 1`] = `
<table>
  <thead>
    <tr>
      <td>
        uuid
      </td>
      <td>
        gender
      </td>
      <td>
        age
      </td>
      <td>
        longitude
      </td>
      <td>
        rate
      </td>
      <td>
        latitude
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        f853578c-fc0f-4e65-81b8-566c5dffa35a
      </td>
      <td>
        F
      </td>
      <td>
        39
      </td>
      <td>
        -0.09998975
      </td>
      <td>
        14.21
      </td>
      <td>
        51.75436293
      </td>
    </tr>
    <tr>
      <td>
        fe2c3195-aeff-487a-a08f-e0bdc0ec6e9a
      </td>
      <td>
        M
      </td>
      <td>
        66
      </td>
      <td>
        0.18228006
      </td>
      <td>
        39.5
      </td>
      <td>
        51.74640997
      </td>
    </tr>
  </tbody>
</table>
`;
